core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/LimitedIterableTest.java

```diff
@@ -27,20 +27,22 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
 import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertTrue;
 
-public class LimitedCloseableIterableTest {
+public class LimitedIterableTest {
 
     @Test
     public void shouldLimitResultsToFirstItem() {
         final List<Integer> values = Arrays.asList(0, 1, 2, 3);
         final int start = 0;
         final int end = 1;
 
-        final CloseableIterable<Integer> limitedValues = new LimitedCloseableIterable<>(values, start, end);
-
-        assertThat(limitedValues).containsExactlyElementsOf(values.subList(start, end));
+        LimitedIterable<Integer> limitedValues = null;
+        try {
+            limitedValues = new LimitedIterable<>(values, start, end);
+            assertThat(limitedValues).containsExactlyElementsOf(values.subList(start, end));
+        } finally {
+            CloseableUtil.close(limitedValues);
```

Comment:
Not sure whether it was required since the change or it was down to good/bad habit

Reply:
Removed the `try-finally` blocks from the tests.

Reply:
This makes sense to me now, tests don't need to finally call close because they are short lived and exceptions shouldn't happen.
In operational code, if an exception might happen that would potentially leave a Closeable un-closed, they should be closed in a finally block to prevent a memory leak.
This is the same as how it should have been done before.
The only difference is, now you don't sometimes get a `CloseableIterable` type returned (and therefore know you must finally close it), instead you should ensure you close the `Iterable` if it may have come from an AccumuloStore and be Closeable. To do this, you use the CloseableUtil in Gaffer.